Lib/logging/__init__.py

```diff
@@ -1543,9 +1558,17 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        try:
+            return self._cache[level]
+        except KeyError:
+            _acquireLock()
+            if self.manager.disable >= level:
```

Comment:
I understand why the `manager.disable` check is in `Logger.isEnabledFor()`. What I was trying to say is having `LoggerAdapter.isEnabledFor()` reimplement `Logger.isEnabledFor()` only makes sense if someone is going to subclass `LoggerAdapter` and overwrite `LoggerAdapter.getEffectiveLevel()`.

I wasn't associating the caching with the levels, but rather the `isEnabledFor()` call. I guess another benefit of moving it would be it also speeds up calls to `getEffectiveLevel()`. I'm not sure how often that's used outside of this module, but I'm sure it is.